train_tacotron.py

```diff
@@ -101,57 +101,35 @@ def create_gta_features(model: Tacotron,
 
 
 def create_align_features(model: Tacotron,
-                          train_set: DataLoader,
-                          val_set: DataLoader,
                           paths: Paths,
-                          pitch_max_freq: float,
-                          silence_threshold: float,
-                          silence_prob_shift: float) -> None:
+                          config: Dict[str, Any]) -> None:
+
     assert model.r == 1, f'Reduction factor of tacotron must be 1 for creating alignment features! ' \
                          f'Reduction factor was: {model.r}'
     model.eval()
-    device = next(model.parameters()).device  # use same device as model parameters
-    iters = len(val_set) + len(train_set)
-    dataset = itertools.chain(train_set, val_set)
-    att_score_dict = {}
-    duration_extractor = DurationExtractor(silence_threshold=silence_threshold,
-                                           silence_prob_shift=silence_prob_shift)
-    sum_att_score = 0
+    model.decoder.prenet.train()
 
-    print('Extracting durations using dijkstra...')
-    for i, batch in enumerate(dataset, 1):
-        batch = to_device(batch, device=device)
-        x, mel = batch['x'], batch['mel']
-        with torch.no_grad():
-            _, _, att_batch = model(x, mel)
-
-        x = batch['x'][0].cpu()
-        mel_len = batch['mel_len'][0].cpu()
-        item_id = batch['item_id'][0]
-        mel = batch['mel'][0, :, :mel_len].cpu()
-        att = att_batch[0, :mel_len, :].cpu()
-
-        # we use the standard alignment score and the more accurate attention score from the duration extractor
-        align_score, _ = attention_score(att_batch, batch['mel_len'], r=1)
-        align_score = float(align_score[0])
-        durs, att_score = duration_extractor(x=x, mel=mel, att=att)
-        durs = np_now(durs).astype(np.int)
-        att_score_dict[item_id] = (align_score, att_score)
-        sum_att_score += att_score
-
-        if np.sum(durs) != mel_len:
-            print(f'WARNINNG: Sum of durations did not match mel length for item {item_id}!')
-        np.save(str(paths.alg / f'{item_id}.npy'), durs, allow_pickle=False)
-        bar = progbar(i, iters)
-        msg = f'{bar} {i}/{iters} Files. Avg attention score: {sum_att_score / i} '
-        stream(msg)
+    durex_conf = config['duration_extraction']
+
+    duration_extractor = DurationExtractor(silence_threshold=durex_conf['silence_threshold'],
+                                           silence_prob_shift=durex_conf['silence_prob_shift'])
+
+    duration_extraction_pipe = DurationExtractionPipeline(paths=paths, config=config,
+                                                          duration_extractor=duration_extractor)
+
+    print('Extracting attention matrices from tacotron...')
+    #duration_extraction_pipe.extract_attentions(model, max_batch_size=durex_conf['max_batch_size'])
 
+    num_workers = durex_conf['num_workers']
```

Comment:
I would rename "durex_conf" as it reminds of a certain brand :)

Reply:
Lol, renamed.